src/walk.rs

```diff
@@ -78,6 +78,8 @@ pub fn scan(path_vec: &[PathBuf], pattern: Arc<Regex>, config: Arc<FdOptions>) -
         .git_exclude(config.read_vcsignore)
         .overrides(overrides)
         .follow_links(config.follow_links)
+        // Same file system is only supported on Unix and Windows platforms
+        .same_file_system(config.same_file_system && (cfg!(unix) || cfg!(windows)))
```

Comment:
I don't think that this is the best place to add the `cfg!(unix) || cfg!(windows)` part.

Ideally, we would disable the availability of that option completely for unsupported platforms.

If that doesn't work, I'd rather add this part to `main.cpp`.

Reply:
It's not pretty, but I've adjusted the code so that the option only exists on Unix and Windows platforms.